src/frontend/web_application/src/components/Modal/index.jsx

```diff
@@ -1,40 +1,48 @@
-import React from 'react';
+import React, { PureComponent } from 'react';
 import PropTypes from 'prop-types';
 import ReactModal from 'react-modal';
 import classnames from 'classnames';
 import Button from '../Button';
 import './style.scss';
 
-const Modal = ({ className, title, children, onClose, ...props }) => (
-  <ReactModal
-    className={classnames('m-modal', className)}
-    overlayClassName="m-modal__overlay"
-    {...props}
-  >
-    <header className="m-modal__header">
-      {title && (
-        <div className="m-modal__title">{title}</div>
-      )}
-      <Button className="m-modal__close" onClick={onClose} icon="remove" />
-    </header>
-    <div className="m-modal__content">
-      {children}
-    </div>
-  </ReactModal>
-);
+class Modal extends PureComponent {
+  static propTypes = {
+    className: PropTypes.string,
+    title: PropTypes.node,
+    children: PropTypes.node,
+    onClose: PropTypes.func,
+  };
 
-Modal.propTypes = {
-  className: PropTypes.string,
-  title: PropTypes.node,
-  children: PropTypes.node,
-  onClose: PropTypes.func,
-};
+  static defaultProps = {
+    className: undefined,
+    title: undefined,
+    children: undefined,
+    onClose: () => {},
+  };
 
-Modal.defaultProps = {
-  className: undefined,
-  title: undefined,
-  children: undefined,
-  onClose: () => {},
-};
+  render() {
+    const { className, title, children, onClose, ...props } = this.props;
+    const rootElement = document.getElementById('root');
```

Comment:
express server will throw an error here (it executes componentWillMount and render)

Reply:
(it doesn't because we didn't make it work correctly for some reasons : translation provider and may be some other components)